core/src/banking_stage/thread_aware_account_locks.rs

```diff
@@ -0,0 +1,663 @@
+use {
+    solana_sdk::pubkey::Pubkey,
+    std::{
+        collections::HashMap,
+        fmt::Display,
+        ops::{BitAndAssign, Sub},
+    },
+};
+
+pub const MAX_THREADS: usize = 64;
+
+/// Identifier for a thread
+pub type ThreadId = usize; // 0..MAX_THREADS-1
+
+/// A bit-set of threads an account is scheduled or can be scheduled for.
+#[derive(Copy, Clone, PartialEq, Eq)]
+#[repr(transparent)]
```

Comment:
why repr(transparent) is needed? is there some abi concern outside this pr?

Reply:
Not 100% sure it's needed, I had just put this to guarantee that the ThreadSet is simply a u64 w/ different methods.

I am explicitly checking the size of the struct down where I format as a way so formatting doesn't get stale:
```rust
static_assertions::assert_eq_size!(ThreadSet, u64); // Formatting must change if size changes
```

Reply:
in that case, how about removing it? `assert_eq_size!` seems to work even wihtout `repr(transparent)`. the motto here is strip-things-until-really-needed.

Reply:
done: e6d91335d0ba439c58868aacfb9473aa3f53a37c